cli/azd/internal/agent/tools/dev/command_executor.go

```diff
@@ -121,19 +121,18 @@ func (t CommandExecutorTool) Call(ctx context.Context, input string) (string, er
 	}
 
 	if req.Command == "azd" {
-		blockedCommands := []string{"up", "provision", "deploy", "down"}
-
-		for _, blocked := range blockedCommands {
-			if req.Args[0] == blocked && (len(req.Args) < 2 || req.Args[1] != "--preview") {
-				errorResponse := common.ErrorResponse{
-					Error:   true,
-					Message: "azd command is not supported",
-				}
-
-				jsonData, _ := json.MarshalIndent(errorResponse, "", "  ")
-				return string(jsonData), nil
+		// Ensure --no-prompt is included in args
+		hasNoPrompt := false
+		for _, arg := range req.Args {
+			if arg == "--no-prompt" {
+				hasNoPrompt = true
+				break
 			}
 		}
+
+		if !hasNoPrompt {
+			req.Args = append(req.Args, "--no-prompt")
+		}
```

Comment:
Can you help me understand this?  Are we forcing `--no-prompt` ?

Reply:
And could we also simplify this to `if !slices.Contains(req.Args, "--no-prompt")`?

Reply:
Yes, if our own agent wants to call back into `azd` we are forcing it to always pass it to run with `--no-prompt` so it doesn't get stuck in a prompt flow that it cannot answer.

Reply:
Does it mean that, if user runs `azd foo` and this fails and starts the AI-troubleshooting agent trying to make some fixes, would eventually try to call `azd foo --no-prompt` to try again?   (preventing a new instance of the AI error troubleshooting and also interactive?)   But at the same time, would it change the user's expectations for the command?